apollo-router/src/query_planner/execution.rs

```diff
@@ -200,66 +213,63 @@ impl PlanNode {
-                        let fut = deferred_node.execute(
-                            parameters,
-                            parent_value,
-                            sender.clone(),
-                            &primary_sender,
-                            &mut deferred_fetches,
-                        );
+                        let (primary_sender, _) = tokio::sync::broadcast::channel::<Value>(1);
 
-                        futures.push(fut);
-                    }
+                        for deferred_node in deferred {
+                            let fut = deferred_node
+                                .execute(
+                                    parameters,
+                                    parent_value,
+                                    sender.clone(),
+                                    &primary_sender,
+                                    &mut deferred_fetches,
+                                )
+                                .in_current_span();
 
-                    tokio::task::spawn(
-                        async move {
-                            join_all(futures).await;
+                            futures.push(fut);
                         }
-                        .in_current_span(),
-                    );
-
-                    value = parent_value.clone();
-                    errors = Vec::new();
-                    let span = tracing::info_span!("primary");
-                    if let Some(node) = node {
-                        let (v, _subselect, err) = node
-                            .execute_recursively(
-                                &ExecutionParameters {
-                                    context: parameters.context,
-                                    service_factory: parameters.service_factory,
-                                    schema: parameters.schema,
-                                    supergraph_request: parameters.supergraph_request,
-                                    deferred_fetches: &deferred_fetches,
-                                    options: parameters.options,
-                                    query: parameters.query,
-                                },
-                                current_dir,
-                                &value,
-                                sender,
-                            )
-                            .instrument(span.clone())
-                            .in_current_span()
-                            .await;
-                        let _guard = span.enter();
-                        value.deep_merge(v);
-                        errors.extend(err.into_iter());
-                        subselection = primary_subselection.clone();
 
-                        let _ = primary_sender.send(value.clone());
-                    } else {
-                        let _guard = span.enter();
+                        tokio::task::spawn(async move {
+                            join_all(futures).await;
+                        });
```

Comment:
Previously this was `in_current_span`, why was that removed? (I imagine because we were "double counting" with `instrument`, but I want to make sure I understand.)

Reply:
Discussed with @BrynCooke . The `in_current_span()` was redundant.

Reply:
@garypen and I discussed this offline. There are several interesting points.
* A span is parented upon create - not first entry.
* A span cannot close before all children are closed. If this is desired then `follow-on` must be used.

We had code that called:
```rust
.instrument(span) // The new span
.in_current_span() // The current span at invocation
```
The future gets wrapped by a 2 wrapper that set the span.
When the future is polled the outer wrapper set the span, but then it is immediately set again by the inner wrapper. So there is no point in the outer wrapper.